model.js

```diff
@@ -0,0 +1,327 @@
+var TETRIS = TETRIS || {};
+
+TETRIS.Model = (function(){
+
+  var refresh = function() {
+    var current = _getCurrentPiece();
+    _incrementPiece(current);
+    _checkPosition(current);
+    _clearRows();
+  }
+
+  var blocks = [];
+
+  var getBlocks = function() {
+    return blocks;
+  }
+
+  var _getCurrentPiece = function() {
+    return blocks.slice(0, 4);
+  }
+
+  var _pieceTypes = ['square', 
+                    'bar', 
+                    'leftL', 
+                    'rightL', 
+                    'leftS', 
+                    'rightS']
+
+  var newPiece = function() {
+    // randomly select type of piece
+    var i = Math.floor(Math.random() * 6);
+    switch (_pieceTypes[i]) {
+      case 'square':
+        _squarePiece();
+        break;
+      case 'bar':
+        _barPiece();
+        break;
+      case 'leftL':
+        _leftLPiece();
+        break;
+      case 'rightL':
+        _rightLPiece();
+        break;
+      case 'leftS':
+        _leftSPiece();
+        break;
+      case 'rightS':
+        _rightSPiece();
+        break;
+    }
+  }
+
+  var _squarePiece = function() {
```

Comment:
Overall, this looks great! I can't seem to get the rotation to work though :O What key am I supposed to press?